### PR TITLE
Add edit profile and change password forms

### PR DESCRIPTION
**Pull request by tyanhan**
# Description
Added edit profile and change password forms on the profile page

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# Screenshot
profile

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

File: frontend/components/molecules/MProfilePhoto.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Grid, Box, Button } from '@material-ui/core';
+import Image from 'next/image';
+import profilePicture from '../../public/profilepicture.svg';
+
+const ProfilePhoto = () => {
+  return (
+    <Grid container direction="row" wrap="nowrap">
+      <Grid>
+        <Image src={profilePicture} alt="profile picture" />
```
Comment (MarcusTXK) [inline frontend/components/molecules/MProfilePhoto.tsx]: Update the src to either use the avatar from the backend or to use dicebear's randomly generated profile picture based on their username.

src={avatar || `https://avatars.dicebear.com/api/bottts/${username}.svg`}

File: frontend/components/organisms/OChangePasswordForm.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { Box, Paper, Grid, Divider, Button } from '@material-ui/core';
+import { useState, ChangeEvent } from 'react';
+import MPasswordField from '../molecules/MPasswordField';
+
+const OChangePasswordForm = () => {
+  const [currPass, setCurrPass] = useState('');
+  const [newPass, setNewPass] = useState('');
+  const [confirmPass, setConfirmPass] = useState('');
```
Comment (MarcusTXK) [inline frontend/components/organisms/OChangePasswordForm.tsx:8]: Add validation when they submit to check if these two are the same, if not warn the user. You can refer to `OLoginDialog`.

File: frontend/pages/profile.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const ProfilePage = () => {
+  const [page, setPage] = useState('edit_profile');
+
+  const handleClickEditProfile = () => {
+    setPage('edit_profile');
+  };
+
+  const handleClickChangePassword = () => {
+    setPage('change_password');
+  };
```
Comment (MarcusTXK) [inline frontend/pages/profile.tsx]: Consider using CamelCase for the values.

**Review by MarcusTXK — CHANGES_REQUESTED** [review]
Wow, really good job! Looks really good.  Just some minor changes and it can be merged.

Some things to note:
Please use the thinner versions for the inputs (pass the prop `size="small"`) as the current ones are very thick. Also be mindful when editing shared components, such as changing their heights, as it breaks other components as seen here in the login modal, where the inputs are now of different sizes:
image

Run `yarn lint` and `yarn build` and try to resolve all the errors
image

**Comment by tyanhan** [comment]
I have added the validation to check if the new password is the same as the confirm password. I have also changed to input text boxes size to small and changed the picture such that it displays the image that was uploaded(or the randomly generated image if no file has been uploaded yet).

**Review by MarcusTXK — APPROVED** [review]
LGTM, great job!

**Pull request merged**